lib/spack/spack/test/cmd/env.py

```diff
@@ -1921,6 +1921,140 @@ def test_env_include_concrete_remove_env():
     assert test2.path not in lockfile_as_dict["include_concrete"].keys()
 
 
+def configure_reuse_true(env):
+    with env:
+        config("add", "concretizer:reuse:true")
+
+
+def configure_reuse_from_environment(env):
+    env.manifest.configuration.update(
+        {"concretizer": {"reuse": {"from": [{"type": "environment"}]}}}
+    )
+    env.manifest.changed = True
+    env.write()
+
+
+def configure_reuse_from_environment_test1(env):
+    env.manifest.configuration.update(
+        {"concretizer": {"reuse": {"from": [{"type": {"environment": "test1"}}]}}}
+    )
+    env.manifest.changed = True
+    env.write()
+
+
+def configure_reuse_from_environment_external_test(combined_env):
+    # Create a new environment called external_test
+    env("create", "external_test")
+    ext = ev.read("external_test")
+    with ext:
+        add("mpich@1.0 +debug")
+    ext.concretize()
+    ext.write()
+
+    combined_env.manifest.configuration.update(
+        {
+            "concretizer": {"reuse": {"from": [{"type": {"environment": "external_test"}}]}},
+            "packages": {"mpich": {"require": ["+debug"]}},
```

Comment:
What does this requirement do for the test? I don't understand -- the reuse should get `+debug` anyway because that's what's available.

Reply:
The default is no debug, I am using this to create a new spec in an external environment and make sure that it is picked up when I am only requiring `+debug` in the combined environment. If it failed to reuse, it would select a  3.0 version with debug

Reply:
The default is `~debug`. I am distinguishing here between the `external_test` version of mpich and the `test1` version that is included concretely.

I will add a comment to that effect here.